### PR TITLE
T 072733: Update multipart form request serializer

### PR DESCRIPTION
**Pull request by bconway99**
- [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines*](https://github.com/mindbody/conduit/blob/master/CONTRIBUTING.md).

This pull request includes (pick all that apply):

- [x] Bugfixes
- [ ] New features
- [x] Breaking changes **(Potentially)**
- [ ] Documentation updates
- [ ] Unit tests
- [ ] Other

### Summary
- With the existing implementation we were having difficulties sending **HTTP multipart requests** to the `RestAPI`.
- The issue appeared to be the boundary value that is sent alongside these requests.
- The `RestAPI` endpoint in question _(`/Documents/LiabilityRelease?consumerId=[ID]`)_ wasn't accepting the existing boundary value.
- In a multipart request, a boundary acts like a marker for each key/value pair in the request's form data. 

### Implementation
- This updates the serializer to remove additional boundary content.
- This was done to ensure the endpoint responds successfully.
- In summary this would only affect **HTTP multipart requests** that leverage `MultipartFormRequestSerializer`.

### Test Plan
- I've marked this as including **breaking changes**.
- While this is a minor change I don't have visibility of all services/endpoints which leverage `MultipartFormRequestSerializer`.
- This could potentially break access to certain endpoints if they are expecting the original boundary value.
- If CI or any endpoints break for anyone please let me know.

### CODE REVIEW

File: Sources/Conduit/Networking/Serialization/MultipartFormRequestSerializer.swift [hunk 1]
```diff
@@ -72,7 +72,7 @@ public final class MultipartFormRequestSerializer: HTTPRequestSerializer {
             return String(characters)
         }
 
-        return "----------------------------\(randomCharacters.joined())"
+        return randomCharacters.joined()
```
Comment (plflanagan) [inline Sources/Conduit/Networking/Serialization/MultipartFormRequestSerializer.swift:75]: This function seems to be used only once, to set the `contentBoundary`.
1. Since it is only used there, could we just put the function into a lazy initialization of that variable?
2. `contentBoundary` itself is used in `inlineContentBoundary` and `finalContentBoundary`, which don't seem like they are ever used. Could we delete those while you are here?

**Review by plflanagan — COMMENTED** [review]
I've left one comment, but I will also leave final approval to others who have more experience with conduit and understand the implications better of changing it.

File: Sources/Conduit/Networking/Serialization/MultipartFormRequestSerializer.swift (hunk 1 above)
Comment (bconway99) [reply Sources/Conduit/Networking/Serialization/MultipartFormRequestSerializer.swift]: 1. Good idea and all updated.
2. I might be misunderstanding your comment here because`inlineContentBoundary` & `finalContentBoundary` are used in the function `makeHTTPBody`.

File: Sources/Conduit/Networking/Serialization/MultipartFormRequestSerializer.swift [hunk 2]
```diff
@@ -26,7 +26,19 @@ public final class MultipartFormRequestSerializer: HTTPRequestSerializer {
...
 
     private var formData: [FormPart] = []
-    private let contentBoundary = MultipartFormRequestSerializer.randomContentBoundary()
+
+    lazy var contentBoundary: String = {
+        let letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+        let lettersLength = UInt32(letters.count)
+
+        let randomCharacters = (0..<12).map { _ -> String in
+            let offset = Int(arc4random_uniform(lettersLength))
```
Comment (asolovev) [inline Sources/Conduit/Networking/Serialization/MultipartFormRequestSerializer.swift:35]: Conduit defines its Swift version to be 5.0, so we can use [SE-0202 "Random Unification"](https://github.com/apple/swift-evolution/blob/master/proposals/0202-random-unification.md).
Unfortunately, it returns optionals. So it would be something like `return String((0..<12).map { _ in letters.randomElement() ?? "a" })`

**Review by asolovev — APPROVED** [review]
Overall seems fine. Added a suggestion about using "Random Unification" methods.
There is also a note in [RFC2046](https://www.ietf.org/rfc/rfc2046.txt)
```
   WARNING TO IMPLEMENTORS:  The grammar for parameters on the Content-
   type field is such that it is often necessary to enclose the boundary
   parameter values in quotes on the Content-type line.  This is not
   always necessary, but never hurts.
```
It doesn't seem necessary in our case, but if there will be additional problems, we can try it as well or use a longer boundary.
